pingcap/pd
# server: support peer address

Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve? 
We plan to provide a heterogeneous engine with tikv, which may need to use different addresses to provide kv and raft services. Now kvproto has an optional `peer_address` to support this (https://github.com/pingcap/kvproto/pull/462 https://github.com/pingcap/kvproto/pull/464). 

The corresponding PD modification is mainly updating kvproto.


### What is changed and how it works?
- update kvproto.
- add peer address in the clone option.
- update api doc

### Check List 

Tests 
 - Manual test (add detailed scripts or steps below)
check the output of store API.

Code changes
 - Has HTTP API interfaces change
 - Has persistent data change

Related changes
 - Need to update the documentation
